File: csharp/NVGIFTest.cs
using System;
using System.Drawing;        // for Bitmap
using System.Drawing.Imaging; // for ImageFormat
using System.IO;

class Program
{
    static void Main(string[] args)
    {
        byte[] bytes = File.ReadAllBytes("../images/drawing.nvg");
        Bitmap bmp = NVGIF.NVGIF.Decode(bytes);
        bmp.Save("decoded.png", ImageFormat.Png);
        Console.WriteLine("NVGIF decoded!");
    }
}

File: c/java/NVGIFTest.java
import java.awt.image.BufferedImage;
import java.io.File;
import javax.imageio.ImageIO;
import nvgif.NVGIFDecoder;

public class NVGIFTest {
    public static void main(String[] args) {
        if (args.length == 0) {
            System.err.println("Usage: java NVGIFTest <file.nvg>");
            System.exit(1);
        }

        String filename = args[0];

        try {
            // Decode NVGIF into a BufferedImage
            BufferedImage img = NVGIFDecoder.decode(filename);

            // Save as PNG
            File outFile = new File("output.png");
            ImageIO.write(img, "png", outFile);

            System.out.println("Decoded NVGIF image saved as " + outFile.getAbsolutePath());
            System.out.println("Width: " + img.getWidth());
            System.out.println("Height: " + img.getHeight());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}